### PR TITLE
fix(TaskSelection): reset component filter when updating the page controller properties

### LINKED ISSUE #1141: Review-Indikator: Filter wird nicht zurückgesetzt
## Erwartetes Verhalten

Klickt man auf den Report-Indikator, werden bereits vorhandene Filter unter Analysis zurückgesetzt und die erforderlichen hinzugefügt.
## Momentanes Verhalten

Zeigt man gefilterte Einträge unter Analysis an und klickt anschliessend auf den Report-Indikator, werden Queryparams zwar angepasst, die Filter werden aber nicht zurückgesetzt. Neue Filter durch den Report-Indikator kommen hinzu, bereits vorhandene werden nicht entfernt.

Die Einträge werden korrekt nach dem Report-Indikator gefiltert, die Filter-Anzeige stimmt aber nicht mehr überein.
## Schritte zum Reproduzieren

Analysis:

-    Zuerst ein "reset" der Filterkriterien
-    Dann mache ich einen eigenen Select, z.B. alle Stunden vom selben Kunden/Projekt/Task (ohne weitere Einschränkung)
-    Dann clicke ich auf das Dreieck für die hängigen Reviews -> Die Anzeige der Recordliste verändert sich, ich sehe alle zu reviewenden Tasks.
-    Was jetzt passiert ist, dass zB meine Task-Einschränkung immer noch in den Filterkriterien zu sehen ist aber die Records in der Trefferliste sind immer noch die vom "Dreieck".

### PR DESCRIPTION
**Pull request by MitanOmar**
prepare fix #1141

### CODE REVIEW

File: app/components/task-selection/component.js [hunk 1]
```diff
@@ -128,25 +127,25 @@ export default class TaskSelectionComponent extends Component {
...
 
   /**
    * The manually selected task
    *
    * @property {Task} _task
    * @private
    */
-  @tracked
+  @localCopy("args.initial.task")
   _task = null;
```
Comment (derrabauke) [inline app/components/task-selection/component.js:149]: Move those above the constructor, so it's more obvious that this is locale state.

File: app/components/task-selection/component.js [hunk 2]
```diff
@@ -128,25 +127,25 @@ export default class TaskSelectionComponent extends Component {
...
   /**
    * The manually selected project
    *
    * @property {Project} _project
    * @private
    */
-  @tracked
-  _project = null;
+  @localCopy("args.initial.project")
+  _project = this.args.initial.project || null;
```
Comment (derrabauke) [inline app/components/task-selection/component.js:140]: Why`|| null` ?

File: app/components/task-selection/component.js (hunk 1 above)
Comment (derrabauke) [inline app/components/task-selection/component.js:149]: Same. Why ` || null` ?

File: app/components/task-selection/component.js (hunk 2 above)
Comment (MitanOmar) [reply app/components/task-selection/component.js:140]: because it was like that before my changes

Comment (derrabauke) [reply app/components/task-selection/component.js:140]: it's not necessary any more I guess

File: app/components/task-selection/component.js [hunk 3]
```diff
@@ -22,6 +22,33 @@ export default class TaskSelectionComponent extends Component {
...
   @service tracking;
 
+  /**
+   * The manually selected customer
+   *
+   * @property {Customer} _customer
+   * @private
+   */
+  @localCopy("args.initial.customer")
+  _customer = this.args.initial.customer;
```
Comment (derrabauke) [inline app/components/task-selection/component.js:32]:
```suggestion
  @localCopy("args.initial.customer")
  _customer;
```

You don't need to assign the argument manually, [as you can see in the docs](https://github.com/tracked-tools/tracked-toolbox?tab=readme-ov-file#localcopy).

File: app/components/task-selection/component.js [hunk 4]
```diff
@@ -22,6 +22,33 @@ export default class TaskSelectionComponent extends Component {
...
+
+  /**
+   * The manually selected project
+   *
+   * @property {Project} _project
+   * @private
+   */
+  @localCopy("args.initial.project")
+  _project = this.args.initial.project;
```
Comment (derrabauke) [inline app/components/task-selection/component.js:41]: Same as above

File: app/components/task-selection/component.js [hunk 5]
```diff
@@ -22,6 +22,33 @@ export default class TaskSelectionComponent extends Component {
...
+
+  /**
+   * The manually selected task
+   *
+   * @property {Task} _task
+   * @private
+   */
+  @localCopy("args.initial.task")
+  _task = this.args.initial.task;
```
Comment (derrabauke) [inline app/components/task-selection/component.js:50]: Same as above

**Review by derrabauke — CHANGES_REQUESTED** [review]

**Review by derrabauke — APPROVED** [review]
:rocket:

**Pull request merged**